org.opentosca.bus.management.api.osgievent/src/org/opentosca/bus/management/api/osgievent/MBEventHandler.java

```diff
@@ -157,8 +222,518 @@ public void handleEvent(final Event event) {
+
+        return result;
+    }
+
+    private Collection<String> toStringCollection(String data, String separator) {
+        Collection<String> result = new ArrayList<String>();
+
+        if(data == null || data.isEmpty()) {
+            return result;
+        }
+        
+        String[] split = data.split(separator);
+
+        for (String part : split) {
+            if (part != null && !part.equals("") && !part.isEmpty()) {
+                result.add(part);
+            }
+        }
+
+        return result;
+    }
+
+    private String toCSV(Collection<String> strings) {
+        StringBuilder strB = new StringBuilder();
+
+        strings.forEach(s -> strB.append(s).append(","));
+
+        return strB.toString();
+    }
+
+    private Set<PlanInstanceInput> toPlanInstanceInputs(Map<String, String> inputs) {
+        Set<PlanInstanceInput> result = new HashSet<PlanInstanceInput>();
+
+        for (String key : inputs.keySet()) {
+            String val = inputs.get(key);
+            result.add(new PlanInstanceInput(key, val, "string"));
+        }
+
+        return result;
+    }
+
+    private Map<String, String> createInput(BPELPlan plan) {
+        Collection<String> inputs = plan.getWsdl().getInputMessageLocalNames();
+
+        Map<String, String> result = new HashMap<String, String>();
+
+        for (String input : inputs) {
+            result.put(input, null);
+        }
+
+        return result;
+    }
+
+    private BpelPlanEnginePlugin getBpelDeployPlugin() {
+        BundleContext ctx = Activator.ctx;
+        try {
+            ServiceReference<?>[] refs =
+                ctx.getAllServiceReferences(IPlanEnginePlanRefPluginService.class.getName(), null);
+            if (refs != null) {
+                for (ServiceReference<?> ref : refs) {
+                    IPlanEnginePlanRefPluginService plugin = (IPlanEnginePlanRefPluginService) ctx.getService(ref);
+                    if (plugin instanceof BpelPlanEnginePlugin) {
+                        return (BpelPlanEnginePlugin) plugin;
+                    }
+                }
+            }
+        }
+        catch (InvalidSyntaxException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+            return null;
+        }
+        return null;
+    }
+
+    private ICoreEndpointService getEndpointService() {
+        BundleContext ctx = Activator.ctx;
+        try {
+            ServiceReference<?>[] refs = ctx.getAllServiceReferences(ICoreEndpointService.class.getName(), null);
+            if (refs != null) {
+                for (ServiceReference<?> ref : refs) {
+                    ICoreEndpointService plugin = (ICoreEndpointService) ctx.getService(ref);
+                    return plugin;
+                }
+            }
+        }
+        catch (InvalidSyntaxException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+            return null;
+        }
+        return null;
```

Comment:
Why is this not directly injected into the component? Is it not available at construction time?

Reply:
I could use a component.xml under OSGI-INF to use the dependency injection, but we remove that stuff in another branch right?

Reply:
It doesn't make much of a difference for pulling the changes into the branch where OSGI is removed. In fact, an existing instance variable makes the process slightly easier, because reasoning about the timing of dependency resolution is simplified.